snarkos/network/mod.rs

```diff
@@ -234,23 +234,28 @@ pub async fn handle_listener<N: Network>(listener: TcpListener, ledger: Arc<Ledg
 
 /// Send a ping to all peers every 10 seconds.
 pub async fn send_pings<N: Network>(ledger: Arc<Ledger<N>>) -> Result<(), Box<dyn std::error::Error>> {
-    loop {
-        thread::sleep(time::Duration::from_secs(10));
+    tokio::spawn(async move {
+        let mut interval = tokio::time::interval(time::Duration::from_secs(10));
+        loop {
+            interval.tick().await;
 
-        let peers = ledger.peers().read().clone();
+            let peers = ledger.peers().read().clone();
 
-        for (addr, outbound) in peers.iter() {
-            if let Err(err) = outbound.send(Message::<N>::Ping).await {
-                warn!("Error sending ping {} to {}", err, addr);
+            for (addr, outbound) in peers.iter() {
+                if let Err(err) = outbound.send(Message::<N>::Ping).await {
```

Comment:
```suggestion
                if let Err(err) = outbound.try_send(Message::<N>::Ping) {
```
if the channel is full, we likely want to skip that peer